Algorithms/Dynamic-Programming/Sliding_Window.cpp

```diff
@@ -0,0 +1,56 @@
+//Name : Shivam Singhal
+//Email ID : shivamsinghal0610@gmail.com
+
+//program to calculate the maximum consecutive sum possible
+//in an array such that it is less than or equal to a given 
+//number m
+
+#include <iostream>
+using namespace std;
+ 
+int main() {
+	int n, m;
+	cout << "Please enter the total number of elements :\n"; 
+	cin >> n ;
+	int a[n];
+	cout << "Please enter array elements :\n"; 
+	for(int i=0; i<n ;i++)
+		cin >> a[i];
+	cout << "Please enter maximum sum required:\n"; 
+	cin >> m;
+	int cur_sum=a[0], max_sum=0, start=0;
+	for(int i=1; i<n ;i++){
+		//checking if current sum is less than or equal to m or not, 
+		//and reducing unnecessary elements that occur before index 
+		//being checked if sum is greater than m
+		while(cur_sum > m && start < i){
+			cur_sum = cur_sum - a[start]; 
```

Comment:
Write an explanation whether this works for negative numbers as well? Since you are subtraction a[start], what if it was negative, the current sum would further increase. 
Please provide a clear explanation how are negative numbers handled.

Reply:
I have considered that all entered numbers will be positive. Should I mention that in the explanation ?

Reply:
Can the solution be modified to work for negative numbers as well using **Dynamic Programming** approach?

Reply:
Yes, okay.

Reply:
Okay then, include both the approaches, make another function to handle negative numbers as well keeping this as well. 
Write comments to explain which function handles which cases, and run both of them on appropriate inputs (from user) to demonstrate.

Reply:
I don't think a separate function will be needed. I will just modify line 27 to cur_sum = cur_sum - abs(a[start]);

Reply:
Okay, then explain it in comments, why your algorithm returns the correct answer in both the cases.

Reply:
Okay, I'll add those.

Reply:
@sahilbansal17 I have made the changes.